opencog/guile/SchemeSmobTV.cc

```diff
@@ -460,5 +460,32 @@ SCM SchemeSmob::ss_tv_get_value (SCM s)
 	return SCM_EOL;
 }
 
+/**
+ * Return the truth value mean
+ */
+SCM SchemeSmob::ss_tv_get_mean(SCM s)
+{
+	TruthValue *tv = verify_tv(s, "cog-tv-mean");
+	return scm_from_double(tv->getMean());
+}
```

Comment:
I think you're missing the point: the scheme code will be FASTER than the C++ code. Here's why.  One call across the guile-to-scheme interface takes about 60 microseonds (on my machine, yours will be different) Lets assume that call returns all 3 TV values (mean confidence count).  Unpacking those three values from the a-list takes about a microscond each, so grand total 63 microseconds.   By contrast, if you do three calls across the C++-guile interface to get three single floats, it will take you 60+60+60 == 180 microseconds. Its a bad deal.  Do as much as possible in scheme; minimize boundary crossings.